### ISSUE TITLE
AbstractHighlightingTest does not work correctly 

### ISSUE THREAD

**Issue by HBuender** [eclipse/xtext-eclipse#980]
The testhighlighting method does not work as expected. 

```@Test
def void testHighlighting(){
'''
 entity Abcde{
   property a : String
   property b : String
   property c : String
   property d : String 
   property e : String
   property f : String
}
'''.testHighlighting("Abcde",entityTextStyle)		
}
```

The text "Abcde" should be highlighted in red according to the entityTextStyle (computed in xtext.workshop.basic.ui.highlighting.EntitySemanticHighlighter). To reproduce the problem import the attached projects and execute the xtext.workshop.basic.ui.tests.SemanticHighlightingTest in the project xtext.workshop.basic.ui.tests. 

[semanticHighlighlingTestProblem.zip](https://github.com/eclipse/xtext-eclipse/files/2854741/semanticHighlighlingTestProblem.zip)

**Comment by miklossy** [comment]
I can reproduce the problem. Seems that the [current implementation](https://github.com/eclipse/xtext-eclipse/blob/792c4c62d75f86be1dca66a8d26282d0a43218ab/org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractHighlightingTest.xtend#L112) is not strong enough to wait for the semantic highlighting to be completed before asserting the result.

Hint: adding the following lines to your SemanticHighlightingTest class makes the test pass:
```
  protected override openInEditor(IFile dslFile) {
    val editor = dslFile.openEditor
    
    /*
     * wait for the Xtext framework HighlightingPresenter.updatePresentation()
     * to apply the semantic highlighting executed asynchronously
     */
    while(Display.^default.readAndDispatch) {}
  
    editor.internalSourceViewer.textWidget
  }
```

**Comment by cdietrich** [comment]
Fixed in 2.18
